packages/cog/src/cog/cog.cutline.ts

```diff
@@ -0,0 +1,205 @@
+
+    const imgBounds = quadKeyToBounds(quadKey);
+    const imgPoly: Polygon = {
+        type: 'Polygon',
+        coordinates: GeoJson.toPositionPolygon(imgBounds.scaleFromCenter(1.2).toBbox()),
+    };
+    const polygons = collectPolygons(cutline && cutline.features, imgPoly);
+
+    if (polygons.length == 1 && polygons[0] === imgPoly) cutline = null; // no need to cut images for this quadKey
+
+    const useTifs = new Set<string>();
+    const usePolys = new Set<Polygon>();
+
+    let inputTotal = 0,
+        cropped = 0;
+
+    for (const path of job.source.files) {
+        ++inputTotal;
+        const tiffName = basename(path);
+        const tiffBounds = sourceMap[tiffName];
+
+        if (tiffBounds == null) throw new Error(`can't find ${tiffName} in vrt`);
+
+        if (!imgBounds.intersects(tiffBounds.bounds)) continue; // image outside quadKey
+
+        if (cutline == null) {
+            useTifs.add(path);
+            continue;
+        }
+
+        const foundp: Polygon[] = [];
+
+        for (const p of polygons) {
+            if (booleanWithin(tiffBounds.poly, p)) {
+                useTifs.add(path);
+                foundp.length = 0;
+                break;
+            }
+            if (!booleanDisjoint(tiffBounds.poly, p)) foundp.push(p);
+        }
+
+        if (foundp.length != 0) {
+            ++cropped;
+            useTifs.add(path);
+            for (const p of foundp) usePolys.add(p);
+        }
+    }
+
+    job.source.files = Array.from(useTifs.values());
+
+    if (usePolys.size == 0) job.output.cutline = undefined;
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
+    else job.output.cutline = await Cutline.writeCutline(makeCutline(Array.from(usePolys.values())), tmpFolder);
+
+    for (const b of vrt.tags('VRTRasterBand')) {
+        const children: VNodeElement[] = [];
+        for (const elm of b.elementChildren()) {
+            if (elm.tag === 'SimpleSource' || elm.tag === 'ComplexSource') {
+                const sf = elm.tags('SourceFilename').next().value!;
+                const path = sf.textContent;
+                if (!useTifs.has(path)) continue;
+            }
+            children.push(elm);
+        }
+        b.children = children;
+    }
+
+    logger.info({ inputTotal, outputTotal: useTifs.size + cropped, cropped, polygons: usePolys.size }, 'Tiff count');
+
+    return vrt;
+}
+
+/**
+ * Write a tempory cutline FeatureCollection to geojson file
+ *
+ * @param cutline defines the cutline
+ * @param tmpFolder the directory write the cutline to
+ * @return the path of the cutline file
+ */
+async function writeCutline(cutline: FeatureCollection, tmpFolder: string): Promise<string> {
+    const target = FileOperator.join(tmpFolder, 'cutline.geojson');
+
+    await FileOperator.create(target).write(target, Buffer.from(JSON.stringify(cutline)));
+
+    return target;
+}
+
+const Cutline = {
+    buildVrt,
+    writeCutline,
+};
+
+export default Cutline;
```

Comment:
avoid default exports, they suck for refactoring + typescript auto imports.

just `export const Cutline ...`